lib/client/identity.go

```diff
@@ -0,0 +1,97 @@
+package client
+
+import (
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/gravitational/teleport/lib/auth/native"
+	"github.com/gravitational/teleport/lib/utils"
+	"github.com/gravitational/trace"
+)
+
+// MakeNewKey generates a new unsigned key. Such key must be signed by a
+// Teleport CA (auth server) before it becomes useful.
+func MakeNewKey() (key *Key, err error) {
+	key = &Key{}
+	keygen := native.New()
+	defer keygen.Close()
+	key.Priv, key.Pub, err = keygen.GenerateKeyPair("")
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	return key, nil
+}
+
+// IdentityFileFormat describes possible file formats how a user identity can be sotred
+type IdentityFileFormat string
+
+const (
+	// IdentityFormatFile is when a key + cert are stored concatenated into a single file
+	IdentityFormatFile IdentityFileFormat = "file"
+
+	// IdentityFormatDir is OpenSSH-compatible format, when a key and a cert are stored in
+	// two different files (in the same directory)
+	IdentityFormatDir IdentityFileFormat = "dir"
+
+	// DefaultIdentityFormat is what Teleport uses by default
+	DefaultIdentityFormat = IdentityFormatFile
+)
+
+// MakeIdentityFile takes a username + his credentials and saves them to disk
+// in a specified format
+func MakeIdentityFile(username, fp string, key *Key, format IdentityFileFormat) (err error) {
+	const (
+		// the files and the dir will be created with these permissions:
+		fileMode = 0600
+		dirMode  = 0770
```

Comment:
tbh I would set modes to 0600 and 0700 because to restrict any external access regardless umask